### ISSUE TITLE
Can't type in textfields when running from iOS simulator

### ISSUE THREAD

**Issue by smallketchup82** [ppy/osu#27984]
### Type

Game behaviour

### Bug description

When running osu! in an iOS simulator, you can't type into textfields. Keybinds like ppp still work, so input is being detected. Typing on other apps like safari works, so it doesn't seem to be an issue with how my xcode/keyboard is set up.

Simulator specs:
- iPhone 15
- iOS 17.4
- Xcode 15.3

My Macbook's specs:
<img width="348" alt="image" src="https://github.com/ppy/osu/assets/69545310/466cc6cd-33bd-4793-9c40-1883f15e9ece">

Running osu! master on commit a35bf77

### Version

a35bf77

### Logs

Since this is an iOS simulator, I can't get the logs folder. I've included the debug logs from the console output
[logs.txt](https://github.com/ppy/osu/files/15096162/logs.txt)

**Comment by Susko3** [comment]
Relevant bit from the logs (not sure what it means):

```cs
2024-04-24 10:40:21.292299-0400 osu.iOS[54572:4257604] [HardwareKeyboard] -[UIApplication getKeyboardDevicePropertiesForSenderID:shouldUpdate:usingSyntheticEvent:], failed to fetch device property for senderID (778835616971358209) use primary keyboard info instead.
2024-04-24 10:40:21.305208-0400 osu.iOS[54572:4257604] [HardwareKeyboard] -[UIApplication getKeyboardDevicePropertiesForSenderID:shouldUpdate:usingSyntheticEvent:], failed to fetch device property for senderID (778835616971358209) use primary keyboard info instead.
```

Could you enable SDL3 event logging and then post the output. Just add the following line to the [main program](https://github.com/ppy/osu/blob/29a6349fbe564dd3cdd8b238cf052e1869222dfa/osu.iOS/Application.cs#L11):

```cs
SDL3.SDL_SetHint(SDL3.SDL_HINT_EVENT_LOGGING, "1"u8);
```

**Comment by smallketchup82** [comment]
@Susko3 Doesn't seem like I can do that. `1>Application.cs(12,13): Error CS0103 : The name 'SDL3' does not exist in the current context`. There's no option to reference it from somewhere else either..? I can do the same thing but change it to `            SDL2.SDL.SDL_SetHint(SDL2.SDL.SDL_HINT_EVENT_LOGGING, "1"u8.ToString());`

**Comment by Susko3** [comment]
You shouldn't have access to SDL2 on the latest `master`, the SDL3 update has already been merged. Please delete `bin` and `obj` folders (they should be in `osu.iOS/`) and then clean and rebuild your solution and project. Also ensure you're on the latest master commit.

**Comment by smallketchup82** [comment]
Right, I was accidentally on a different branch when I tested it earlier. Going back to the commit listed in the original comment, I still get a `System.DllNotFoundException: SDL3` when i put `SDL3.SDL_SetHint(SDL3.SDL_HINT_EVENT_LOGGING, "1"u8);` in `Application.cs`

**Comment by smoogipoo** [comment]
You should put that hint in SDL3Window. Check the existing code there for where you might put it.

**Comment by Susko3** [comment]
Just put it anywhere in the game, like the constructor of `OsuGame`, if you don't have a local checkout of framework (required for SDL3Window change).

**Comment by smallketchup82** [comment]
Got it, I put it in the load() function of `OsuGame`. Here are the logs:
[log2.txt](https://github.com/ppy/osu/files/15113237/log2.txt)

**Comment by smallketchup82** [comment]
Hold on, I forgot to enable hardware keyboard. Disregard the above logs
[log3.txt](https://github.com/ppy/osu/files/15113272/log3.txt)

**Comment by Susko3** [comment]
I presume you're typing with your mac keyboard. What if you instead try to type with the on-screen keyboard in iOS (if that option is available)?

I've created an upstream issue: https://github.com/libsdl-org/SDL/issues/9624. Feel free to subscribe to it.

**Comment by smallketchup82** [comment]
The on screen keyboard doesn't show, and if I disable hardware keyboard (so that it cannot recognize my macbook keyboard), the software keyboard doesn't show, neither automatically nor manually

**Comment by frenzibyte** [comment]
This will now work as expected once https://github.com/libsdl-org/SDL/pull/11406 is in.

**Comment by frenzibyte** [comment]
Fixed since https://github.com/ppy/osu-framework/pull/6486
